app/routes/downloadPaymentsByYearSummary.js

```diff
@@ -0,0 +1,40 @@
+const { Parser } = require('json2csv')
+const { getSchemePaymentsByYear } = require('../services/databaseService')
+
+module.exports = {
+  method: 'GET',
+  path: '/downloadPaymentsByYearSummary',
+  handler: async (_request, _response) => {
+    const fields = [
+      'financial_year',
+      'scheme',
+      'amount'
+    ]
+    try {
+      const schemePaymentsArray = []
+      const csvParser = new Parser({ fields })
+      const schemePayments = (await getSchemePaymentsByYear()).reduce((acc, item) => {
+        (acc[item.financial_year] = acc[item.financial_year] || []).push(item)
+        return acc
+      }, {})
+
+      for (const property in schemePayments) {
+        for (const item of schemePayments[property]) {
+          item.amount = item.total_amount
+          delete item.total_amount
+          schemePaymentsArray.push(item)
+        }
+      }
+
+      const csv = csvParser.parse(schemePaymentsArray)
```

Comment:
I believe the loops in the try block can be simplified like below:
```
      const csvParser = new Parser({ fields })
      const schemePaymentsByYear = (await getSchemePaymentsByYear()).sort((r1, r2) => r1.financial_year > r2.financial_year ? 1 : -1)
      const schemePayments = schemePaymentsByYear.map(({ total_amount, ...rest }) => ({
        amount: total_amount,
        ...rest
      }))
      
      const csv = csvParser.parse(schemePayments)
```

Please take a look and let me know what you think.

Reply:
🤯 Amazing! Thanks